### ISSUE TITLE
Bug: lingva engine / https://lingva.ml/ is offline

### ISSUE THREAD

**Issue by return42** [searxng/searxng#2718]
**Version of SearXNG, commit number if you are using on master branch and stipulate if you forked SearXNG**

Repository: https://github.com/searxng/searxng
Branch: master
Version: 2023.7.19+a446dea1b

**How did you install SearXNG?**

`make run`

**Technical report**

Error
  * Error: httpx.ConnectError
  * Percentage: 100
  * Parameters: `(None, None, 'lingva.ml')`
  * File name: `searx/search/processors/online.py:118`
  * Function: `_send_http_request`
  * Code: `response = req(params['url'], **request_args)`

----

Alternatives

- https://github.com/thedaviddelta/lingva-translate#instances

**Comment by Bnyro** [comment]
Ref https://fosstodon.org/@thedaviddelta/110069243482314227 and
https://fosstodon.org/@thedaviddelta/11006924852795449.

Looks like they have been target of ddos attacks or api abusage, which is why the instance probably got taken down due to too much traffic that exceeds their hosting plan.

Apart from that, .ml domains are provided by Freenom which has been shut down recently. Source: 
https://krebsonsecurity.com/2023/05/phishing-domains-tanked-after-meta-sued-freenom/

Additionally, the development seems to be discontinued, hence I don't expect any new official instance anytime soon.

Tl;Dr
Imo, we should cycle through the working public instances like we do for other services like Piped or Invidious.

**Comment by return42** [comment]
> Apart from that, .ml domains are provided by Freenom which has been shut down recently.

Good to know :+1: 

> Imo, we should cycle through the working public instances like we do for other services like Piped or Invidious.

Yes I tried to implement a PR that cycles .. but sadly the [instances](https://github.com/thedaviddelta/lingva-translate#instances) have different APIs .. seems some does not have an API.

Not sure what we can do further ..

**Comment by Bnyro** [comment]
I found [this comment](https://github.com/thedaviddelta/lingva-translate/commit/8e45af1396bad30917c1d508f08ddd78ef0589fc#commitcomment-58565534), of which the second and third url work and seem to be official:
* https://lingva-translate-thedaviddelta.vercel.app/
* https://lingva-translate-git-main-thedaviddelta.vercel.app/

They feel a bit slow however at first glance.

I personally use lingva.lunar.icu, but I doubt it's a good idea to set that as the main instance.

**Comment by Bnyro** [comment]
lingva.ml seems to be functional again, so this can be closed.

**Comment by Bnyro** [comment]
Oh, but it's no longer official: https://fosstodon.org/@thedaviddelta/111376540936289966. So we should likely switch to the new official instance as soon as possible.

**Comment by return42** [comment]
> Oh, but it's no longer official: https://fosstodon.org/@thedaviddelta/111376540936289966. So we should likely switch to the new official instance as soon as possible.

@Bnyro  I send https://github.com/searxng/searxng/pull/3072 with the new official URL / if you have time, you can take a look at it :+1:
